File: public/docs/_examples/dependency-injection/ts/app/car-factory.ts
import {Engine, Tires, Car} from './car';

// #docregion
class SuperFactory {

  createEngine() {
    return new Engine();
  }

  createTires() {
    return new Tires();
  }

  createCar() {
    return new Car(this.createEngine(), this.createTires());
  }
}
// #enddocregion

File: public/docs/_examples/dependency-injection/dart/lib/car/car_factory.dart
// #docregion
import 'car.dart';

// BAD pattern!
class CarFactory {
  Car createCar() =>
      new Car(createEngine(), createTires())
        ..description = 'Factory';

  Engine createEngine() => new Engine();
  Tires createTires() => new Tires();
}
